File: api/internal/model/git.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type GitConnection struct {
	ID            uuid.UUID `json:"id"`
	AppID         uuid.UUID `json:"app_id"`
	Provider      string    `json:"provider"`
	RepoURL       string    `json:"repo_url"`
	RepoOwner     string    `json:"repo_owner"`
	RepoName      string    `json:"repo_name"`
	Branch        string    `json:"branch"`
	AutoDeploy    bool      `json:"auto_deploy"`
	WebhookSecret string    `json:"-"`
	AccessToken   string    `json:"-"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: web/src/api/git.ts
import client from './client'

export interface GitConnection {
  id: string
  appId: string
  provider: 'github' | 'gitlab'
  repoUrl: string
  branch: string
  autoDeploy: boolean
  connectedAt: string
}

export interface ConnectRepoPayload {
  provider: 'github' | 'gitlab'
  repoUrl: string
  branch: string
  accessToken: string
}

export function connectRepo(appId: string, payload: ConnectRepoPayload) {
  return client.post<GitConnection>(`/apps/${appId}/git`, payload)
}

export function disconnectRepo(appId: string) {
  return client.delete(`/apps/${appId}/git`)
}

export function getGitConnection(appId: string) {
  return client.get<GitConnection>(`/apps/${appId}/git`)
}

export function toggleAutoDeploy(appId: string, enabled: boolean) {
  return client.patch<GitConnection>(`/apps/${appId}/git`, { autoDeploy: enabled })
}
